.github/workflows/e2e.yml

```diff
@@ -0,0 +1,49 @@
+name: E2E staging
+on:
+  push:
+    branches:
+      - changeset-release/*
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        saleor: [318, 319]
+    env:
+      ACCESS_TOKEN: ${{ secrets.SALEOR_TOKEN }}
+      SALEOR_VERSION: ${{ matrix.saleor }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Saleor CLI
+        run: |
+          jq --null-input \
+             --arg token "Token $ACCESS_TOKEN" \
+             '{"token":$token,"telemetry":"false","saleor_env":"production","cloud_api_url":"https://cloud.saleor.io/platform/api","organization_slug":"shopex","organization_name":"Shopex"}' > ~/.config/saleor.json
+      - name: Setup PNPM
+        uses: pnpm/action-setup@a3252b78c470c02df07e9d59298aecedc3ccdd6d # v3.0.0
+        with:
+          run_install: |
+            - args: [--frozen-lockfile, --filter=app-avatax]
+      - name: Get Saleor snapshot
+        run: |
+          BACKUP=$(pnpm dlx saleor backup list --name="snapshot-avatax-$SALEOR_VERSION" --latest --json)
+          BACKUP_ID=$(echo "$BACKUP" | jq -r '.[0].key')
+          echo "BACKUP_ID=$BACKUP_ID" >> "$GITHUB_ENV"
+      - name: Restore Saleor snapshot
+        run: |
+          pnpm dlx saleor backup restore --from="$BACKUP_ID" --environment="shopex-avatax-$SALEOR_VERSION" --skip-webhooks-update
+      - name: Generate GraphQL files
+        # TODO: Switch to 1pass item after https://linear.app/saleor/issue/SHOPX-303
+        env:
+          TEST_SALEOR_API_URL: "https://shopex-avatax-${{ env.SALEOR_VERSION }}.eu.saleor.cloud/graphql/"
+        run: pnpm --filter=app-avatax generate
+      - name: Run e2e tests
+        env:
+          TEST_SALEOR_API_URL: "https://shopex-avatax-${{ env.SALEOR_VERSION }}.eu.saleor.cloud/graphql/"
+        run: pnpm --filter=app-avatax e2e
+      # TODO: Add HTML report: https://linear.app/saleor/issue/SHOPX-304
```

Comment:
I guess we hardcode avatax now, but in general we want to run every app here

I also wonder, do we really want to have separate saleor envs for each app? Why don't we have a single env (shopex-319...) and use different channels for different apps?

cc @szczecha

Reply:
https://linear.app/saleor/issue/SHOPX-305/adjust-apps-monorepo-cicd-to-run-e2e-for-every-app

Reply:
Yes - but it requires at least two apps with e2e 😄 - I'm wondering if in the future we will have at least 3 workflows:
1. Run e2e for avatax
2. Run e2e for e.g emails-messages
3. Run all e2e

Workflows 1 & 2 can have different setups - e.g avatax uses pactum and vitests and emails-messages may use playwright.